db/api/src/settings.rs

```diff
@@ -0,0 +1,230 @@
+use crate::{db::HoprDb, errors::Result};
+use async_trait::async_trait;
+use hopr_internal_types::prelude::AliasEntry;
+use sea_orm::{ActiveModelTrait, ColumnTrait, EntityTrait, QueryFilter};
+
+pub const ALIASES_ENTRY_NAME: &str = "aliases";
+
+/// Defines DB API for accessing HOPR settings (mainly aliases for now)
+#[async_trait]
+pub trait HoprDbSettingsOperations {
+    /// Retrieve the alias for a given PeerID
+    async fn get_alias(&self, alias: String) -> Result<Option<String>>;
+
+    /// Retrive all aliases
+    async fn get_aliases(&self) -> Result<Vec<AliasEntry>>;
+
+    /// Create new key pair value in the db
+    async fn set_alias(&self, peer: String, alias: String) -> Result<bool>;
+
+    /// Update aliases
+    async fn set_aliases(&self, aliases: Vec<AliasEntry>) -> Result<bool>;
+
+    /// Delete alias
+    async fn delete_alias(&self, alias: String) -> Result<bool>;
+}
+
+#[async_trait]
+impl HoprDbSettingsOperations for HoprDb {
+    async fn get_alias(&self, alias: String) -> Result<Option<String>> {
```

Comment:
I would call this something like `resolve_alias` since you are not getting the alias data structure entirely, but the peer id mapping.